### PR TITLE
ORCHESTRATIONS: Add Lab Command

### PR DESCRIPTION
**Pull request by AndrewMcClelland**
Closes Task #40547669

### CODE REVIEW

File: DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        private static LabCommand CreateRandomLabCommand() =>
+            CreateLabCommandFiller(GetRandomDateTimeOffset()).Create();
+
+        private static DateTimeOffset GetRandomDateTimeOffset() =>
+            new DateTimeRange(earliestDate: new DateTime()).GetValue();
+
+        private static LabCommand CreateRandomLabCommand(DateTimeOffset date) =>
+            CreateLabCommandFiller(date).Create();
+
+        private static Filler<LabCommand> CreateLabCommandFiller(DateTimeOffset dateTimeOffset)
```
Comment (hassanhabib) [inline DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.cs:71]: Always move fillers to the bottom of the page.
The methods that used outside of the file takes the higher priority in ordering

File: DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            this.loggingBrokerMock = new Mock<ILoggingBroker>();
+
+            this.labCommandOrchestrationService = new LabCommandOrchestrationService(
+                labCommandService: this.labCommandServiceMock.Object,
+                loggingBroker: this.loggingBrokerMock.Object);
+        }
+
+        public static TheoryData<Xeption> LabCommandDependencyValidationExceptions()
+        {
+            string randomErrorMessage = GetRandomString();
```
Comment (hassanhabib) [inline DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.cs:37]: Either name this `randomMessage` or change the one below - I vote `randomMessage`

File: DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.Exceptions.Add.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+using Xeptions;
+using Xunit;
+
+namespace DMX.Core.Api.Tests.Unit.Services.Orchestrations.LabCommands
+{
+    public partial class LabCommandOrchestrationServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(LabCommandDependencyValidationExceptions))]
+        public async Task ShouldThrowOrchestrationDependencyValidationExceptionOnAddIfDependencyValidationErrorOccursAndLogItAsync(
```
Comment (hassanhabib) [inline DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.Exceptions.Add.cs:20]: Break down to less than 120. Maybe get rid of the word `Exception`

File: DMX.Core.Api/Services/Orchestrations/LabCommands/LabCommandOrchestrationService.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        private readonly ILabCommandService labCommandService;
+        private readonly ILoggingBroker loggingBroker;
+
+        public LabCommandOrchestrationService(
+            ILabCommandService labCommandService,
+            ILoggingBroker loggingBroker)
+        {
+            this.labCommandService = labCommandService;
+            this.loggingBroker = loggingBroker;
+        }
```
Comment (hassanhabib) [inline DMX.Core.Api/Services/Orchestrations/LabCommands/LabCommandOrchestrationService.cs:23]: Give me a new line under this one

File: DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.Logic.Add.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public partial class LabCommandOrchestrationServiceTests
+    {
+        [Fact]
+        public async Task ShouldAddLabCommandAsync()
+        {
+            // given
+            LabCommand randomLabCommand = CreateRandomLabCommand();
+            LabCommand inputLabCommand = randomLabCommand;
+            LabCommand returnedLabCommand = inputLabCommand.DeepClone();
```
Comment (hassanhabib) [inline DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabCommands/LabCommandOrchestrationServiceTests.Logic.Add.cs:22]: Do we need `DeepClone` in here?

**Comment by AndrewMcClelland** [comment]
Pending completion of LabCommandEventService - we can then call the Enqueue method from this orchestration service.

File: DMX.Core.Api/Services/Orchestrations/LabCommands/LabCommandOrchestrationService.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+        public ValueTask<LabCommand> AddLabCommandAsync(LabCommand labCommand) =>
+        TryCatch(async () =>
+        {
+            ValidateLabCommand(labCommand);
+            await this.labCommandEventService.AddLabCommandEventAsync(labCommand);
+
+            return await this.labCommandService.AddLabCommandAsync(labCommand);
```
Comment (hassanhabib) [inline DMX.Core.Api/Services/Orchestrations/LabCommands/LabCommandOrchestrationService.cs:35]:
```csharp
var mockSeqence = new MockSequence();

            this.labCommandServiceMock.InSequence(mockSeqence).Setup(broker =>
                broker.AddLabCommandAsync(inputLabCommand))
                    .ReturnsAsync(returnedLabCommand);

            this.labCommandEventServiceMock.InSequence(mockSeqence).Setup(broker =>
                broker.AddLabCommandEventAsync(inputLabCommand))
                    .ReturnsAsync(eventLabCommand);

```

**Review by hassanhabib — APPROVED** [review]

**Pull request merged**